### PR TITLE
Improve plugins scaffolding for custom fields

### PR DESCRIPTION
**Pull request by dgrammatiko**
#### Summary of Changes

Right now if you want to include some custom fields in your plugin you need to include a custom field path in the relative xml file, something like:

```
<fieldset addfieldpath="/plugins/system/<plugin name>/field">
```

Documented here: https://docs.joomla.org/Creating_a_custom_form_field_type

This PR leverages this, so custom fields will happily reside inside a folder named `field`

Why field and not fields?
Singular word help us transition to autoload and namespace
#### Testing Instructions

Apply this patch with patch tester
Go to plugins tinyMCE and if you can see a dropdown for the skins you have a successful test:
screen shot 2016-04-08 at 00 00 21

Also go to Profile and check the Term of service and Date of birth, if dropdown then success:
screen shot 2016-04-08 at 00 00 05

Check the stats plugin and observe that all fields are correctly rendered/functioning
#### B/C

No backwards compatibility break
#### Performance

We just add one more folder check to the whole stack, so performance degradation should be unnoticeable. (Joomla itself does that hundreds of times for core libraries)

### CODE REVIEW

File: plugins/user/profile/profile.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@
 	</languages>
 	<config>
 		<fields name="params">
-			<fieldset name="basic" addfieldpath="/administrator/components/com_content/models/fields">
```
Comment (mbabker) [inline plugins/user/profile/profile.xml]: Getting antsy on your delete button.  Plugins aren't "autoloading" component field paths.

Comment (dgt41) [inline plugins/user/profile/profile.xml]: Opps...

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 7bcedf1ec2b41b118d9fa631daadbd5ed7218a2e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9796">issues.joomla.org/joomla-cms/9796</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 5dbed4690f3908dda76876047f4dace460ac48dc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9796">issues.joomla.org/joomla-cms/9796</a>.</sub>

**Comment by MATsxm** [comment]
I have tested this item :white_check_mark: successfully on 5dbed4690f3908dda76876047f4dace460ac48dc<br /><br />Thanks :+1: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9796">issues.joomla.org/joomla-cms/9796</a>.</sub>

**Comment by brianteeman** [comment]
RTC - leaving it to the maintainers to decide the milestone<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9796">issues.joomla.org/joomla-cms/9796</a>.</sub>

**Comment by wilsonge** [comment]
Merged into 3.6.x with https://github.com/joomla/joomla-cms/commit/6e4b1923a23ede4ebe7eb5913452cf12c4883750 - thanks @dgt41

**Comment by infograf768** [comment]
@dgt41 
regression in profile edit. see #10615

**Pull request closed without merging**